commit a2815d8a6cb3a1643acc73896b7f07bc8b8244d8
Author: Yang Bo <yangbo@linuxdeepin.com>
Date:   Sun Jan 3 19:15:54 2016 +0800

    UBUNTU: SAUCE: (no-up) kconfig: in debug mode some 0 length message prints occur
    
    Has no real kernel impact, so continue to carry.
    
    When we enable the zconfdump() debugging we see assertion failures
    attempting to print the config.  Convert this into a noop.
    
    Signed-off-by: Andy Whitcroft <apw@canonical.com>
    Signed-off-by: Yang Bo <yangbo@linuxdeepin.com>

diff --git a/scripts/kconfig/lkc.h b/scripts/kconfig/lkc.h
index 91ca126..9faeb23 100644
--- a/scripts/kconfig/lkc.h
+++ b/scripts/kconfig/lkc.h
@@ -88,7 +88,9 @@ void set_all_choice_values(struct symbol *csym);
 /* confdata.c and expr.c */
 static inline void xfwrite(const void *str, size_t len, size_t count, FILE *out)
 {
-	assert(len != 0);
+	//assert(len != 0);
+	if (len == 0)
+		return;
 
 	if (fwrite(str, len, count, out) != count)
 		fprintf(stderr, "Error in writing or end of file.\n");
